src/Core/Utilities/CaptchaProtectedAttribute.cs

```diff
@@ -0,0 +1,37 @@
+using Bit.Core.Context;
+using Bit.Core.Exceptions;
+using Bit.Core.Services;
+using Bit.Core.Models.Api;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Bit.Core.Utilities
+{
+    public class CaptchaProtectedAttribute : ActionFilterAttribute
+    {
+        public string ModelParameterName { get; set; } = "model";
+
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            var currentContext = context.HttpContext.RequestServices.GetRequiredService<ICurrentContext>();
+            var captchaValidationService = context.HttpContext.RequestServices.GetRequiredService<ICaptchaValidationService>();
+
+            if (captchaValidationService.ServiceEnabled && (currentContext.IsBot || captchaValidationService.RequireCaptcha))
+            {
+                var captchaResponse = (context.ActionArguments[ModelParameterName] as ICaptchaProtectedModel)?.CaptchaResponse;
+
+                if (string.IsNullOrWhiteSpace(captchaResponse))
+                {
+                    throw new BadRequestException("HCaptcha_SiteKey", captchaValidationService.SiteKey);
```

Comment:
Did we want to leave the key name more generic vs. component/3rd party specific?

Reply:
Probably the name should be provided by the validation service. That way we can alter how the client presents the captcha depending on the service key provided.